src/Data.Json.JSemantic.purs

```diff
@@ -0,0 +1,105 @@
+module Data.Json.JSemantic
+  ( JSemantic(..)
+  , toSemantic
+  ) where
+
+import Data.Argonaut.JCursor(JsonPrim(..), runJsonPrim)
+import Data.Maybe
+import Data.String.Regex (RegexFlags(..), regex, test)
+import Data.String
+import Data.Foldable
+import qualified Data.Array as A
+import qualified Data.Array.Unsafe as AU
```

Comment:
I'll comment on this below but let's try to see if we can eliminate use of this (it's usually possible and sometimes cleans up the code).